src/System.Data.SqlClient/src/System/Data/SqlClient/SqlCommand.cs

```diff
@@ -1265,7 +1274,13 @@ public IAsyncResult BeginExecuteXmlReader(AsyncCallback callback, object stateOb
                 cachedAsyncState.SetActiveConnectionAndResult(completion, nameof(EndExecuteXmlReader), _activeConnection);
                 if (writeTask != null)
                 {
-                    AsyncHelper.ContinueTask(writeTask, completion, () => BeginExecuteXmlReaderInternalReadStage(completion));
+                    AsyncHelper.ContinueTaskWithState(writeTask, completion,
+                        state: Tuple.Create(this, completion),
+                        onSuccess: state => {
```

Comment:
This shouldn't be closing at all because all the variables unside the lambda are passed through the state variable. I'll disassemble it later and verify that. ContinueTaskWithState is doing to allocate anyway if we're on an async path it's enevitable that we have to put the state somewhere.

Reply:
@Wraith2  Can you verify this with the disassembly? Whatever you find, if it is actionable, then you could take it up in a separate PR. I would like to harden this PR and run validations with EF tests on Linux.

Reply:
It's ok, the original looked as I expected and i'll leave this PR as it is now.